### PR TITLE
Networking configuration in kickstart templates

### PR DESCRIPTION
**Pull request by ares**
extracted from https://github.com/theforeman/foreman/pull/1565

### CODE REVIEW

**Comment by stbenjam** [comment]
Can this be merged for 1.7.2? I'd like to be able to add the networking feature support to Katello 2.1 templates.

File: kickstart/provision.erb [hunk 1]
```diff
@@ -29,7 +29,11 @@ lang en_US.UTF-8
 selinux --enforcing
 keyboard us
 skipx
-network --bootproto <%= @static ? "static --ip=#{@host.ip} --netmask=#{@host.subnet.mask} --gateway=#{@host.subnet.gateway} --nameserver=#{[@host.subnet.dns_primary,@host.subnet.dns_secondary].reject{|n| n.blank?}.join(',')}" : 'dhcp' %> --hostname <%= @host %>
+
+<% subnet = @host.subnet -%>
+<% dhcp = subnet.dhcp_boot_mode? -%>
```
Comment (ekohl) [inline kickstart/provision.erb]: Which version of foreman is required for this?

Comment (GregSutcliffe) [inline kickstart/provision.erb]: 1.7

File: kickstart/provision.erb [hunk 2]
```diff
@@ -29,7 +29,11 @@ lang en_US.UTF-8
 selinux --enforcing
 keyboard us
 skipx
-network --bootproto <%= @static ? "static --ip=#{@host.ip} --netmask=#{@host.subnet.mask} --gateway=#{@host.subnet.gateway} --nameserver=#{[@host.subnet.dns_primary,@host.subnet.dns_secondary].reject{|n| n.blank?}.join(',')}" : 'dhcp' %> --hostname <%= @host %>
```
Comment (ares) [inline kickstart/provision.erb]: @static is now part of dhcp detection on line 34, if it is set to true, dhcp will result in false

File: kickstart/provision.erb (hunk 1 above)
Comment (ekohl) [inline kickstart/provision.erb]: IIRC we had the policy to keep support for stable - 1, so 1.6 in this case. Can this somehow be written to support 1.6?

Comment (ares) [inline kickstart/provision.erb]: I don't think so, or at least other than `if Foreman.version > 1.6 then $magic`, these attributes are not present in 1.6

Comment (domcleal) [inline kickstart/provision.erb]: `#respond_to?` is supported since 1.5.3 (for this reason), if that helps.

**Comment by ares** [comment]
I added conditions with `respond_to?`, tested with Foreman 1.6 it renders fine.

**Comment by ekohl** [comment]
:+1:

**Comment by mmoll** [comment]
works for me :+1:
want to squash the commits? not sure what's better here...

**Comment by ares** [comment]
Whatever you prefer. I can squash everything unless @bcrochet objects.

**Comment by mmoll** [comment]
I'd prefer squashing, @bcrochet ok?

**Comment by bcrochet** [comment]
Squash works for me. :+1:

**Comment by ares** [comment]
squashed

**Comment by mmoll** [comment]
merged as 61cfd65 at last, thanks @ares and @bcrochet! :tada:

**Pull request closed without merging**